### ISSUE TITLE
Process instance with parallel flow can't be completed after interruption by Event sub process

### ISSUE THREAD

**Issue by Zelldon** [camunda-cloud/zeebe#6565]
After fixing  https://github.com/camunda-cloud/zeebe/issues/6552 via https://github.com/camunda-cloud/zeebe/pull/6562 I realized that there are still some models not working with event sub processes.

![diagram_3](https://user-images.githubusercontent.com/2758593/111323006-cb928f80-8669-11eb-8568-cc1e69862572.png)
![diagram_6](https://user-images.githubusercontent.com/2758593/111372127-86d31c80-869a-11eb-8f02-ed565edca14e.png)
![diagram_5](https://user-images.githubusercontent.com/2758593/111372141-8b97d080-869a-11eb-8854-1eb677e25bfd.png)

What do they have in common you may ask? A parallel execution flow when the interrupting event sub process is triggered. It seems that when this is happening the token count is not decremented correctly and then again the process instance cant be completed.

I have an reproducer (which fails most of the time):
```

  @Test
  // https://github.com/camunda-cloud/zeebe/issues/6552
  public void shouldEndProcessWithParallelFlow() {
    // given
    final ProcessBuilder process = Bpmn.createExecutableProcess(PROCESS_ID);

    process
        .eventSubProcess("event_sub_proc")
        .startEvent("event_sub_start")
        .interrupting(true)
        .message(b -> b.name(MSG_NAME).zeebeCorrelationKeyExpression("key"))
        .endEvent("event_sub_end");

    final BpmnModelInstance model =
        process
            .startEvent("start_proc")
            .sequenceFlowId("toParallel")
            .parallelGateway("parallel")
            .intermediateCatchEvent("catch")
            .message(m -> m.name("msg").zeebeCorrelationKeyExpression("key"))
            .endEvent("end_proc")
            .moveToLastGateway()
            .intermediateCatchEvent("catch1")
            .message(m -> m.name("msg1").zeebeCorrelationKeyExpression("key1"))
            .endEvent("end_proc1")
            .done();

    engineRule.deployment().withXmlResource(model).deploy();

    final long wfInstanceKey =
        engineRule
            .processInstance()
            .ofBpmnProcessId(PROCESS_ID)
            .withVariables(Map.of("key", 123, "key1", 123))
            .create();

    // when
    RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.CREATED)
        .withProcessInstanceKey(wfInstanceKey)
        .withMessageName(MSG_NAME)
        .await();
    RecordingExporter.processInstanceRecords()
        .withElementType(BpmnElementType.START_EVENT)
        .withIntent(ProcessInstanceIntent.ELEMENT_ACTIVATING)
        .withElementId("start_proc")
        .withProcessInstanceKey(wfInstanceKey)
        .await();

    engineRule
        .message()
        .withName(MSG_NAME)
        .withCorrelationKey("123")
        .withVariables(Map.of("key", "123"))
        .publish();

    // then
    assertThat(
            RecordingExporter.processInstanceRecords()
                .withProcessInstanceKey(wfInstanceKey)
                .limitToProcessInstanceCompleted())
        .extracting(r -> tuple(r.getValue().getBpmnElementType(), r.getIntent()))
        .containsSubsequence(
            tuple(BpmnElementType.START_EVENT, ProcessInstanceIntent.EVENT_OCCURRED),
            tuple(BpmnElementType.SUB_PROCESS, ProcessInstanceIntent.ELEMENT_ACTIVATED),
            tuple(BpmnElementType.SUB_PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED),
            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));

    Loggers.STREAM_PROCESSING.error(RecordingExporter.getRecords().toString());
  }

```

I tested it with the fix https://github.com/camunda-cloud/zeebe/pull/6562 and without on develop it failed on both sides, so the new fix doesn't caused that.

**Comment by Zelldon** [comment]
```
06:00:58.428 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 0 new token count 1 
 io.zeebe.engine.state.instance.ElementInstance.spawnToken(ElementInstance.java:128)
io.zeebe.engine.state.instance.DbElementInstanceState.spawnToken(DbElementInstanceState.java:192)
io.zeebe.engine.state.appliers.ProcessInstanceElementActivatingApplier.applyState(ProcessInstanceElementActivatingApplier.java:60)
io.zeebe.engine.state.appliers.ProcessInstanceElementActivatingApplier.applyState(ProcessInstanceElementActivatingApplier.java:21)
io.zeebe.engine.state.appliers.EventAppliers.applyState(EventAppliers.java:201)
io.zeebe.engine.processing.streamprocessor.writers.EventApplyingStateWriter.appendFollowUpEvent(EventApplyingStateWriter.java:49)
io.zeebe.engine.processing.streamprocessor.writers.EventApplyingStateWriter.appendFollowUpEvent(EventApplyingStateWriter.java:39)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior.activateChildInstance(BpmnStateTransitionBehavior.java:209)
io.zeebe.engine.processing.bpmn.container.ProcessProcessor.onActivated(ProcessProcessor.java:76)
io.zeebe.engine.processing.bpmn.container.ProcessProcessor.onActivated(ProcessProcessor.java:23)

06:00:58.455 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 1 new token count 2 
 io.zeebe.engine.state.instance.ElementInstance.spawnToken(ElementInstance.java:128)
io.zeebe.engine.state.instance.DbElementInstanceState.spawnToken(DbElementInstanceState.java:192)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.spawnToken(BpmnStateBehavior.java:117)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior.takeSequenceFlow(BpmnStateTransitionBehavior.java:190)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior.lambda$takeOutgoingSequenceFlows$0(BpmnStateTransitionBehavior.java:295)
java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior.takeOutgoingSequenceFlows(BpmnStateTransitionBehavior.java:295)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onCompleted(StartEventProcessor.java:62)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onCompleted(StartEventProcessor.java:20)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:154)

06:00:58.456 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 2 new token count 1 
 io.zeebe.engine.state.instance.ElementInstance.consumeToken(ElementInstance.java:142)
io.zeebe.engine.state.instance.DbElementInstanceState.consumeToken(DbElementInstanceState.java:183)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.consumeToken(BpmnStateBehavior.java:110)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onCompleted(StartEventProcessor.java:63)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onCompleted(StartEventProcessor.java:20)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:154)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processRecord(BpmnStreamProcessor.java:106)
io.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.processRecord(TypedRecordProcessor.java:54)
io.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.lambda$processInTransaction$3(ProcessingStateMachine.java:296)
io.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:84)

06:00:58.467 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 1 new token count 2 
 io.zeebe.engine.state.instance.ElementInstance.spawnToken(ElementInstance.java:128)
io.zeebe.engine.state.instance.DbElementInstanceState.spawnToken(DbElementInstanceState.java:192)
io.zeebe.engine.state.appliers.ProcessInstanceElementActivatingApplier.applyState(ProcessInstanceElementActivatingApplier.java:60)
io.zeebe.engine.state.appliers.ProcessInstanceElementActivatingApplier.applyState(ProcessInstanceElementActivatingApplier.java:21)
io.zeebe.engine.state.appliers.EventAppliers.applyState(EventAppliers.java:201)
io.zeebe.engine.processing.streamprocessor.writers.EventApplyingStateWriter.appendFollowUpEvent(EventApplyingStateWriter.java:49)
io.zeebe.engine.processing.streamprocessor.writers.EventApplyingStateWriter.appendFollowUpEvent(EventApplyingStateWriter.java:39)
io.zeebe.engine.processing.bpmn.behavior.BpmnEventSubscriptionBehavior.publishActivatingEvent(BpmnEventSubscriptionBehavior.java:236)
io.zeebe.engine.processing.bpmn.behavior.BpmnEventSubscriptionBehavior.triggerInterruptingEventSubProcess(BpmnEventSubscriptionBehavior.java:424)
io.zeebe.engine.processing.bpmn.behavior.BpmnEventSubscriptionBehavior.lambda$triggerEventSubProcess$15(BpmnEventSubscriptionBehavior.java:400)

06:00:58.469 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 2 new token count 3 
 io.zeebe.engine.state.instance.ElementInstance.spawnToken(ElementInstance.java:128)
io.zeebe.engine.processing.bpmn.behavior.BpmnEventSubscriptionBehavior.lambda$triggerInterruptingEventSubProcess$16(BpmnEventSubscriptionBehavior.java:434)
io.zeebe.engine.state.instance.DbElementInstanceState.updateInstance(DbElementInstanceState.java:175)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.updateFlowScopeInstance(BpmnStateBehavior.java:69)
io.zeebe.engine.processing.bpmn.behavior.BpmnEventSubscriptionBehavior.triggerInterruptingEventSubProcess(BpmnEventSubscriptionBehavior.java:431)
io.zeebe.engine.processing.bpmn.behavior.BpmnEventSubscriptionBehavior.lambda$triggerEventSubProcess$15(BpmnEventSubscriptionBehavior.java:400)
io.zeebe.engine.processing.bpmn.behavior.BpmnEventSubscriptionBehavior.triggerEvent(BpmnEventSubscriptionBehavior.java:290)
io.zeebe.engine.processing.bpmn.behavior.BpmnEventSubscriptionBehavior.triggerEventSubProcess(BpmnEventSubscriptionBehavior.java:389)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onEventOccurred(StartEventProcessor.java:85)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onEventOccurred(StartEventProcessor.java:20)

06:00:58.505 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.process - Don't process event because of an illegal state transition: Expected flow scope instance to be not interrupted but was interrupted by an event with key '2251799813685261'. [context: {intent=SEQUENCE_FLOW_TAKEN, elementId=toParallel, bpmnElementType=SEQUENCE_FLOW, elementInstanceKey=2251799813685260, flowScopeKey=2251799813685251, processInstanceKey=2251799813685251, parentProcessInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=proc, processVersion=1, processDefinitionKey=2251799813685249}]
06:00:58.512 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 0 new token count 1 
 io.zeebe.engine.state.instance.ElementInstance.spawnToken(ElementInstance.java:128)
io.zeebe.engine.state.instance.DbElementInstanceState.spawnToken(DbElementInstanceState.java:192)
io.zeebe.engine.state.appliers.ProcessInstanceElementActivatingApplier.applyState(ProcessInstanceElementActivatingApplier.java:60)
io.zeebe.engine.state.appliers.ProcessInstanceElementActivatingApplier.applyState(ProcessInstanceElementActivatingApplier.java:21)
io.zeebe.engine.state.appliers.EventAppliers.applyState(EventAppliers.java:201)
io.zeebe.engine.processing.streamprocessor.writers.EventApplyingStateWriter.appendFollowUpEvent(EventApplyingStateWriter.java:49)
io.zeebe.engine.processing.streamprocessor.writers.EventApplyingStateWriter.appendFollowUpEvent(EventApplyingStateWriter.java:39)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior.activateChildInstance(BpmnStateTransitionBehavior.java:209)
io.zeebe.engine.processing.bpmn.container.SubProcessProcessor.onActivated(SubProcessProcessor.java:69)
io.zeebe.engine.processing.bpmn.container.SubProcessProcessor.onActivated(SubProcessProcessor.java:22)

06:00:58.523 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 1 new token count 2 
 io.zeebe.engine.state.instance.ElementInstance.spawnToken(ElementInstance.java:128)
io.zeebe.engine.state.instance.DbElementInstanceState.spawnToken(DbElementInstanceState.java:192)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.spawnToken(BpmnStateBehavior.java:117)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior.takeSequenceFlow(BpmnStateTransitionBehavior.java:190)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior.lambda$takeOutgoingSequenceFlows$0(BpmnStateTransitionBehavior.java:295)
java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior.takeOutgoingSequenceFlows(BpmnStateTransitionBehavior.java:295)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onCompleted(StartEventProcessor.java:62)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onCompleted(StartEventProcessor.java:20)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:154)

06:00:58.524 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 2 new token count 1 
 io.zeebe.engine.state.instance.ElementInstance.consumeToken(ElementInstance.java:142)
io.zeebe.engine.state.instance.DbElementInstanceState.consumeToken(DbElementInstanceState.java:183)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.consumeToken(BpmnStateBehavior.java:110)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onCompleted(StartEventProcessor.java:63)
io.zeebe.engine.processing.bpmn.event.StartEventProcessor.onCompleted(StartEventProcessor.java:20)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:154)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processRecord(BpmnStreamProcessor.java:106)
io.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.processRecord(TypedRecordProcessor.java:54)
io.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.lambda$processInTransaction$3(ProcessingStateMachine.java:296)
io.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:84)

06:00:58.533 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 1 new token count 0 
 io.zeebe.engine.state.instance.ElementInstance.consumeToken(ElementInstance.java:142)
io.zeebe.engine.state.instance.DbElementInstanceState.consumeToken(DbElementInstanceState.java:183)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.consumeToken(BpmnStateBehavior.java:110)
io.zeebe.engine.processing.bpmn.event.EndEventProcessor.onCompleted(EndEventProcessor.java:86)
io.zeebe.engine.processing.bpmn.event.EndEventProcessor.onCompleted(EndEventProcessor.java:26)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:154)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processRecord(BpmnStreamProcessor.java:106)
io.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.processRecord(TypedRecordProcessor.java:54)
io.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.lambda$processInTransaction$3(ProcessingStateMachine.java:296)
io.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:84)

06:00:58.538 [Broker-0-StreamProcessor-1] ERROR io.zeebe.broker.logstreams - Current token count: 3 new token count 2 
 io.zeebe.engine.state.instance.ElementInstance.consumeToken(ElementInstance.java:142)
io.zeebe.engine.state.instance.DbElementInstanceState.consumeToken(DbElementInstanceState.java:183)
io.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.consumeToken(BpmnStateBehavior.java:110)
io.zeebe.engine.processing.bpmn.container.SubProcessProcessor.onCompleted(SubProcessProcessor.java:92)
io.zeebe.engine.processing.bpmn.container.SubProcessProcessor.onCompleted(SubProcessProcessor.java:22)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:154)
io.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processRecord(BpmnStreamProcessor.java:106)
io.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.processRecord(TypedRecordProcessor.java:54)
io.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.lambda$processInTransaction$3(ProcessingStateMachine.java:296)
io.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:84)

06:01:03.580 [] INFO  io.zeebe.test - Test failed, following records were exported:
06:01:03.621 [] INFO  io.zeebe.test - Compact log representation:
--------
	['C'ommand/'E'event/'R'ejection] [valueType] [intent] - #[position]->#[source record position]  K[key] - [summary of value]
	K999 - key; #999 - record position; "ID" element/process id; @"elementid"/[K99] - element with ID and key
	Long numbers are substituted with short numbers (e.g. 52124672368 -> 1). Substituted numbers are used consistently, but they might not have the same order as the numbers they substitute
	Long IDs are shortened (e.g. 'startEvent_5d56488e-0570-416c-ba2d-36d2a3acea78' -> 'star..acea78'
--------
C DEPLOYMENT             CREATE           - #01->#-1 K-1 - process.xml
E PROC                   CREATED          - #02->#01 K00 - process.xml->"proc" (version:1)
E DEPLOYMENT             CREATED          - #03->#01 K01 - process.xml
E DEPLOYMENT             FULLY_DISTR      - #04->#01 K01 - 
C PROC_INSTANCE_CREATION CREATE           - #05->#-1 K-1 - new <process "proc"> with variables: {key1=123, key=123}
E VARIABLE               CREATED          - #06->#05 K02 - key1->123 in <process [K03]>
E VARIABLE               CREATED          - #07->#05 K04 - key->123 in <process [K03]>
E PROC_INSTANCE          ELMNT_ACTIVATING - #08->#05 K03 - PROCESS "proc" in <process "proc"[K03]>
E PROC_INSTANCE_CREATION CREATED          - #09->#05 K05 - new <process "proc"> with variables: {key1=123, key=123}
E PROC_INSTANCE_SUB      CREATING         - #10->#08 K-1 - "messageName" (inter.) correlationKey: 123 @[K03] in <process "proc"[K03]> (no vars)
E PROC_INSTANCE          ELMNT_ACTIVATED  - #11->#08 K03 - PROCESS "proc" in <process "proc"[K03]>
C MSG_SUB                CREATE           - #12->#-1 K-1 - "messageName" (inter.) correlationKey: 123 @[K03] in <process "proc"[K03]> (no vars)
E PROC_INSTANCE          ELMNT_ACTIVATING - #13->#11 K06 - START_EVENT "start_proc" in <process "proc"[K03]>
E MSG_SUB                CREATED          - #14->#12 K07 - "messageName" (inter.) correlationKey: 123 @[K03] in <process "proc"[K03]> (no vars)
C PROC_INSTANCE_SUB      CREATE           - #15->#-1 K-1 - "messageName" (inter.) @[K03] in <process ?> (no vars)
E PROC_INSTANCE          ELMNT_ACTIVATED  - #16->#13 K06 - START_EVENT "start_proc" in <process "proc"[K03]>
E PROC_INSTANCE_SUB      CREATED          - #17->#15 K-1 - "messageName" (inter.) @[K03] in <process ?> (no vars)
C MSG                    PUBLISH          - #18->#-1 K-1 - "messageName" correlationKey: 123 with variables: {key=123}
E PROC_INSTANCE          ELMNT_COMPLETING - #19->#16 K06 - START_EVENT "start_proc" in <process "proc"[K03]>
E MSG                    PUBLISHED        - #20->#18 K08 - "messageName" correlationKey: 123 with variables: {key=123}
E MSG_SUB                CORRELATING      - #21->#18 K07 - "messageName" (inter.) correlationKey: 123 @[K03] in <process "proc"[K03]> with variables: {key=123}
C PROC_INSTANCE_SUB      CORRELATE        - #22->#-1 K-1 - "messageName" (inter.) correlationKey: 123 @[K03] in <process "proc"[K03]> with variables: {key=123}
E PROC_INSTANCE          ELMNT_COMPLETED  - #23->#19 K06 - START_EVENT "start_proc" in <process "proc"[K03]>
E PROC_INSTANCE          EVENT_OCCURRED   - #24->#22 K09 - START_EVENT "event_sub_start" in <process "proc"[K03]>
E PROC_INSTANCE_SUB      CORRELATED       - #25->#22 K-1 - "messageName" (inter.) correlationKey: 123 @[K03] in <process "proc"[K03]> with variables: {key=123}
C MSG_SUB                CORRELATE        - #26->#-1 K-1 - "messageName" (inter.) @[K03] in <process "proc"[K03]> (no vars)
E PROC_INSTANCE          SEQ_FLOW_TAKEN   - #27->#23 K10 - SEQUENCE_FLOW "toParallel" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_ACTIVATING - #28->#24 K11 - SUB_PROCESS "event_sub_proc" in <process "proc"[K03]>
E MSG_SUB                CORRELATED       - #29->#26 K07 - "messageName" (inter.) correlationKey: 123 @[K03] in <process "proc"[K03]> with variables: {key=123}
E PROC_INSTANCE          ELMNT_ACTIVATED  - #30->#28 K11 - SUB_PROCESS "event_sub_proc" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_ACTIVATING - #31->#30 K12 - START_EVENT "event_sub_start" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_ACTIVATED  - #32->#31 K12 - START_EVENT "event_sub_start" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_COMPLETING - #33->#32 K12 - START_EVENT "event_sub_start" in <process "proc"[K03]>
E VARIABLE               UPDATED          - #34->#33 K04 - key->"123" in <process [K03]>
E PROC_INSTANCE          ELMNT_COMPLETED  - #35->#33 K12 - START_EVENT "event_sub_start" in <process "proc"[K03]>
E PROC_INSTANCE          SEQ_FLOW_TAKEN   - #36->#35 K13 - SEQUENCE_FLOW "sequenc..86506c7" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_ACTIVATING - #37->#36 K14 - END_EVENT "event_sub_end" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_ACTIVATED  - #38->#37 K14 - END_EVENT "event_sub_end" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_COMPLETING - #39->#38 K14 - END_EVENT "event_sub_end" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_COMPLETED  - #40->#39 K14 - END_EVENT "event_sub_end" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_COMPLETING - #41->#40 K11 - SUB_PROCESS "event_sub_proc" in <process "proc"[K03]>
E PROC_INSTANCE          ELMNT_COMPLETED  - #42->#41 K11 - SUB_PROCESS "event_sub_proc" in <process "proc"[K03]>
```

**Comment by Zelldon** [comment]
The problem is this https://github.com/camunda-cloud/zeebe/blob/develop/engine/src/main/java/io/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java#L52

Especially this https://github.com/camunda-cloud/zeebe/blob/develop/engine/src/main/java/io/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java#L99-L101 It seems I can't really say that it is an non interrupting event sub process. It just spawns tokens for my interrupting event sub process, which is wrong.
